backend/src/services/kanban/util.ts

```diff
@@ -0,0 +1,97 @@
+import { calculateWorkDaysBy24Hours } from "../common/WorkDayCalculate";
+import { CycleTimeInfo } from "../../contract/kanban/KanbanStoryPointResponse";
+import { CardStepsEnum } from "../../models/kanban/CardStepsEnum";
+import { Issue } from "@linear/sdk";
+import { JiraCard } from "../../models/kanban/JiraCard";
+
+export type StatusChangedArrayItem = {
+  timestamp: number;
+  status: string;
+};
+
+export function getThisStepCostTime(
+  index: number,
+  statusChangedArrayItems: StatusChangedArrayItem[]
+): number {
+  if (index < statusChangedArrayItems.length - 1)
+    return calculateWorkDaysBy24Hours(
+      statusChangedArrayItems[index].timestamp,
+      statusChangedArrayItems[index + 1].timestamp
+    );
+  return calculateWorkDaysBy24Hours(
+    statusChangedArrayItems[index].timestamp,
+    Date.now()
+  );
+}
+
+export function getCardTimeForEachStep(
+  timeLine: StatusChangedArrayItem[]
+): CycleTimeInfo[] {
+  const result = new Map<string, number>();
+  timeLine.forEach((statusChangedArrayItem, index, statusChangedArrayItems) => {
+    const addedTime = result.get(statusChangedArrayItem.status.toUpperCase());
+    const costedTime = getThisStepCostTime(index, statusChangedArrayItems);
+    const value = addedTime
+      ? +(costedTime + addedTime).toFixed(2)
+      : +costedTime.toFixed(2);
+    result.set(statusChangedArrayItem.status.toUpperCase(), value);
+  });
+  const cycleTimeInfos: CycleTimeInfo[] = [];
+  result.forEach(function (value, key) {
+    cycleTimeInfos.push(new CycleTimeInfo(key, value));
+  });
+  return cycleTimeInfos;
+}
+
+export function sortStatusChangedArray(
+  statusChangedArray: StatusChangedArrayItem[]
+): StatusChangedArrayItem[] {
+  return statusChangedArray.sort((a, b) => a.timestamp - b.timestamp);
+}
+
+export function reformTimeLineForFlaggedCards(
+  statusChangedArray: StatusChangedArrayItem[]
+): StatusChangedArrayItem[] {
+  const needToFilterArray: number[] = [];
+  const timeLine = sortStatusChangedArray(statusChangedArray);
+  timeLine.forEach(function (
```

Comment:
It looks a little complicated. Is there a way we can make it less complex?

Reply:
its a simple extraction of the private static method in backend/src/services/kanban/Jira/Jira.ts, the idea is that in case the method gonna be applied to other cycleTime calculations.